denonavr/audyssey.py

```diff
@@ -71,21 +72,20 @@ class DenonAVRAudyssey(DenonAVRFoundation):
     # AppCommand0300.xml interface
     appcommand0300_attrs = {AppCommands.GetAudyssey: None}
 
-    def setup(self) -> None:
+    async def async_setup(self) -> None:
         """Ensure that the instance is initialized."""
         # Add tags for a potential AppCommand.xml update
         for tag in self.appcommand0300_attrs:
             self._device.api.add_appcommand0300_update_tag(tag)
 
-        self._device.telnet_api.register_callback(
-            "PS", self._async_sound_detail_callback
-        )
+        asyncio.create_task(self._async_register_audyssey_callbacks())
```

Comment:
When creating task instead of running function synchronously the overall performance is slower not faster.
You might not notice this here, because this only creates the task for a function and does not wait for its execution. Since the task is referenced nowhere, it is even possible that python garbage collector collects it before it is executed.

Reply:
Yeah, I noticed. I've actually fixed that in my fork, but didn't submit it here, had to make other changes that I don't think you want either. You might want to have a look at the main branch in the fork and see what from it that you'd want.

It's also not the callbacks that causes the slowdown, it's the registration of events on telnet without delays that overwhelms the receiver. Only sending four or five of them locks the receiver up until it has processed everything. I solved this by sorting the registrations by importance and adding a 100ms delay between each registration.

Reply:
What do you mean with `locks the receiver up`? Does it react slowly, is it ignoring commands or is it something else?

Reply:
It locks it up as in not processing any additional commands until it's done with the event responses, resulting in timeouts. This is perhaps not very noticeable in HA where you rarely reboot, but for the remote where this library is used it causes frustration because the remote goes into sleep mode after a few minutes and then the connection is established again when you for example want to change the volume.